pkg/controllers/cloud/cloudentityselector_controller.go

```diff
@@ -94,7 +90,61 @@ func (r *CloudEntitySelectorReconciler) SetupWithManager(mgr ctrl.Manager) error
 
 func (r *CloudEntitySelectorReconciler) processCreateOrUpdate(selector *cloudv1alpha1.CloudEntitySelector,
 	selectorNamespacedName *types.NamespacedName) error {
-	accPoller, preExists := r.addAccountPoller(selector)
+	accountNamespacedName := &types.NamespacedName{
+		Namespace: selector.Namespace,
+		Name:      selector.Spec.AccountName,
+	}
+	r.selectorToAccountMap[*selectorNamespacedName] = *accountNamespacedName
+	accPoller, preExists := GetAccountPoller(accountNamespacedName)
+	if accPoller == nil {
+		err := fmt.Errorf("poller is nil, selector couldn't be added, account %s", accountNamespacedName.String())
+		r.Log.Error(err, "selector add/update failed")
+		return err
+	}
+
+	accPoller.selector = selector.DeepCopy()
```

Comment:
Please add comment here explaining why you need this?

Reply:
added